File: exercism/ruby/bob/bob.rb
class Bob
  def hey (line)
    if /\A^\s*\Z/ =~ line then "Fine. Be that way!"
    elsif /\A[^a-z]*[A-Z][^a-z]*\Z/ =~ line then "Whoa, chill out!"
    elsif line.end_with?("?") then "Sure."
    else "Whatever."
    end
  end
end

File: exercism/kotlin/bob/src/main/kotlin/Bob.kt
object Bob {
  private fun isUpperCase(query: String): Boolean {
    return query.toUpperCase() == query && query.toLowerCase() != query
  }

  fun hey(query: String): String {
    return when {
      query.isBlank() -> "Fine. Be that way!"
      isUpperCase(query) -> "Whoa, chill out!"
      query.endsWith("?") -> "Sure."
      else -> "Whatever."
    }
  }
}
